plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -320,7 +330,11 @@ function CalibreCompanion:sendBook(arg)
     local inbox_dir = G_reader_settings:readSetting("inbox_dir")
     local filename = inbox_dir .. "/" .. arg.lpath
     DEBUG("write to file", filename)
-    local outfile = io.open(filename, "wb")
+    local lfs = require("libs/libkoreader-lfs")
+    local dir, name = string.match(filename, "(.-)([^\\/]-%.?([^%.\\/]*))$")
```

Comment:
Oh, there's a util function for that: http://koreader.rocks/doc/modules/util.html#splitFilePathName